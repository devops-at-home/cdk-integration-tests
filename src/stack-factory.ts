import { App, StackProps } from 'aws-cdk-lib';
import { IotCoreCertificates } from './stacks/iot-core-certificates.stack';

export class StackFactory {
    constructor(app: App, stackProps?: StackProps) {
        new IotCoreCertificates(app, 'iot-core-certs', stackProps);
    }
}
