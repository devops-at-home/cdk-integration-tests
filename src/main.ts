import { App } from 'aws-cdk-lib';
import { StackFactory } from './stack-factory';

const env = {
    account: process.env.CDK_DEFAULT_ACCOUNT,
    region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new StackFactory(app, { env });

app.synth();
