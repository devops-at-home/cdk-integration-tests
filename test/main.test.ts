import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { StackFactory } from '../src/stack-factory';

describe('Snapshot tests', () => {
    const app = new App();

    const { stacks } = new StackFactory(app);

    stacks.forEach((stack) => {
        const template = Template.fromStack(stack);
        expect(template.toJSON()).toMatchSnapshot();
    });
});
