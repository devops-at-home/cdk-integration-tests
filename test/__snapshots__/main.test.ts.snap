// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration test snapshots IntegTest-IotCoreCerts 1`] = `
{
  "Outputs": {
    "OutputCertId": {
      "Value": {
        "Fn::GetAtt": [
          "ThingWithCertlambdaCustomResource28E12CE8",
          "certId",
        ],
      },
    },
    "OutputCertPem": {
      "Value": {
        "Fn::GetAtt": [
          "ThingWithCertlambdaCustomResource28E12CE8",
          "certPem",
        ],
      },
    },
    "OutputPrivKey": {
      "Value": {
        "Fn::GetAtt": [
          "ThingWithCertlambdaCustomResource28E12CE8",
          "privKey",
        ],
      },
    },
    "OutputThingArn": {
      "Value": {
        "Fn::GetAtt": [
          "ThingWithCertlambdaCustomResource28E12CE8",
          "thingArn",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": {
      "DependsOn": [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "f5b6a6d08a8206e67d0b54e2a29abd7e0472d835abc71037de3a93545b5edacc.zip",
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": [
          {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ThingWithCertLambdaExecutionRole25FAB8F4": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "ThingWithCertLambdaExecutionRoleDefaultPolicy9FF9FB4B": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": "arn:aws:logs:*:*:*",
            },
            {
              "Action": "iot:*",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ThingWithCertLambdaExecutionRoleDefaultPolicy9FF9FB4B",
        "Roles": [
          {
            "Ref": "ThingWithCertLambdaExecutionRole25FAB8F4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ThingWithCertlambdaCustomResource28E12CE8": {
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "ThingWithCertlambdaProviderframeworkonEvent497EC884",
            "Arn",
          ],
        },
        "ThingName": "integrationTest",
      },
      "Type": "AWS::CloudFormation::CustomResource",
    },
    "ThingWithCertlambdaFunction9E90C8BA": {
      "DependsOn": [
        "ThingWithCertLambdaExecutionRoleDefaultPolicy9FF9FB4B",
        "ThingWithCertLambdaExecutionRole25FAB8F4",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "130a39ee4f1c6c5dd63f25970467191bddc211e68f0e8310ce1d1b7c267a08ba.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "ThingWithCertLambdaExecutionRole25FAB8F4",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 10,
      },
      "Type": "AWS::Lambda::Function",
    },
    "ThingWithCertlambdaFunctionLogRetention1BE9E451": {
      "Properties": {
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "ThingWithCertlambdaFunction9E90C8BA",
              },
            ],
          ],
        },
        "RetentionInDays": 1,
        "ServiceToken": {
          "Fn::GetAtt": [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "ThingWithCertlambdaProviderframeworkonEvent497EC884": {
      "DependsOn": [
        "ThingWithCertlambdaProviderframeworkonEventServiceRoleDefaultPolicyC133ABFF",
        "ThingWithCertlambdaProviderframeworkonEventServiceRole95EF1588",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "1eabd374284db340b74179e3429008132f5b6b0b7b28d472d852807d7f5f9746.zip",
        },
        "Description": "AWS CDK resource provider framework - onEvent (IntegTest-IotCoreCerts/ThingWithCert/lambdaProvider)",
        "Environment": {
          "Variables": {
            "USER_ON_EVENT_FUNCTION_ARN": {
              "Fn::GetAtt": [
                "ThingWithCertlambdaFunction9E90C8BA",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": {
          "Fn::GetAtt": [
            "ThingWithCertlambdaProviderframeworkonEventServiceRole95EF1588",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "ThingWithCertlambdaProviderframeworkonEventServiceRole95EF1588": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ThingWithCertlambdaProviderframeworkonEventServiceRoleDefaultPolicyC133ABFF": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "ThingWithCertlambdaFunction9E90C8BA",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "ThingWithCertlambdaFunction9E90C8BA",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ThingWithCertlambdaProviderframeworkonEventServiceRoleDefaultPolicyC133ABFF",
        "Roles": [
          {
            "Ref": "ThingWithCertlambdaProviderframeworkonEventServiceRole95EF1588",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ThingWithCertparamStoreCertPem15876E53": {
      "Properties": {
        "Name": "/devices/integrationTest/certPem",
        "Type": "String",
        "Value": {
          "Fn::GetAtt": [
            "ThingWithCertlambdaCustomResource28E12CE8",
            "certPem",
          ],
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "ThingWithCertparamStorePrivKey2E5E90A6": {
      "Properties": {
        "Name": "/devices/integrationTest/privKey",
        "Type": "String",
        "Value": {
          "Fn::GetAtt": [
            "ThingWithCertlambdaCustomResource28E12CE8",
            "privKey",
          ],
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
